scalding-core/src/main/scala/com/twitter/scalding/typed/OptimizationRules.scala

```diff
@@ -545,12 +527,28 @@ object OptimizationRules {
     }
   }
 
+  /**
+   * This rule is important in that it allows us to reduce
+   * the number of nodes in the graph, which is helpful to speed up rule application
+   */
   object ComposeDescriptions extends PartialRule[TypedPipe] {
+    def combine(descs1: List[(String, Boolean)], descs2: List[(String, Boolean)]): List[(String, Boolean)] = {
+      val combined = descs1 ::: descs2
+      val unordered = combined.collect { case (d, true) => d }.toSet
```

Comment:
this is unused?

Reply:
Nope. I somehow forgot it and implemented without it below. Good catch.